### PR TITLE
♻️ Refactor/liquidation thresholds

### PR DESCRIPTION
**Pull request by Thomas-Smets**

### CODE REVIEW

File: src/AssetRegistry/AbstractPricingModule.sol [hunk 1]
```diff
@@ -128,12 +128,11 @@ abstract contract PricingModule is Ownable {
      * @notice Returns the risk variable arrays of an asset
      * @param asset The address of the asset
      * @return assetCollateralFactors The array of collateral factors for the asset
-     * @return assetLiquidationThresholds The array of liquidation thresholds for the asset
+     * @return assetLiquidationFactors The array of liquidation thresholds for the asset
```
Comment (j-vp) [inline src/AssetRegistry/AbstractPricingModule.sol:131]: + update comments: single collateral factor & liquidation factor is returned
```suggestion
     * @return assetLiquidationFactors The liquidation factor for the asset
```

File: src/AssetRegistry/AbstractPricingModule.sol [hunk 2]
```diff
@@ -190,11 +189,7 @@ abstract contract PricingModule is Ownable {
     function _setRiskVariables(address asset, uint256 basecurrency, RiskVars memory riskVars) internal virtual {
         require(riskVars.collateralFactor <= RiskConstants.MAX_COLLATERAL_FACTOR, "APM_SRV: Coll.Fact not in limits");
 
-        require(
-            riskVars.liquidationThreshold <= RiskConstants.MAX_LIQUIDATION_THRESHOLD
-                && riskVars.liquidationThreshold >= RiskConstants.MIN_LIQUIDATION_THRESHOLD,
-            "APM_SRV: Liq.Thres not in limits"
-        );
+        require(riskVars.liquidationFactor <= RiskConstants.MAX_LIQUIDATION_FACTOR, "APM_SRV: Liq.Thres not in limits");
```
Comment (j-vp) [inline src/AssetRegistry/AbstractPricingModule.sol:192]:
```suggestion
        require(riskVars.liquidationFactor <= RiskConstants.MAX_LIQUIDATION_FACTOR, "APM_SRV: Liq.Fact not in limits");
```

`+ tests

File: src/AssetRegistry/MainRegistry.sol [hunk 3]
```diff
@@ -466,54 +466,30 @@ contract MainRegistry is Ownable {
         RiskModule.AssetValueAndRiskVariables[] memory valuesAndRiskVarPerAsset =
             getListOfValuesPerAsset(assetAddresses, assetIds, assetAmounts, baseCurrency);
 
-        collateralValue = RiskModule.calculateWeightedCollateralValue(valuesAndRiskVarPerAsset);
+        collateralValue = RiskModule.calculateCollateralValue(valuesAndRiskVarPerAsset);
     }
 
     /**
-     * @notice Calculate the liquidation threshold given the asset details in given baseCurrency
+     * @notice Calculate the collateralValue given the asset details in given baseCurrency
```
Comment (j-vp) [inline src/AssetRegistry/MainRegistry.sol:473]:
```suggestion
     * @notice Calculate the liquidationValue given the asset details in given baseCurrency
```

File: src/AssetRegistry/MainRegistry.sol [hunk 4]
```diff
@@ -466,54 +466,30 @@ contract MainRegistry is Ownable {
...
-    function getCollateralValueAndLiquidationThreshold(
+    function getLiquidationValue(
         address[] calldata assetAddresses,
         uint256[] calldata assetIds,
         uint256[] calldata assetAmounts,
         address baseCurrency
-    ) public view returns (uint256 collateralValue, uint256 liquidationThreshold) {
-        //No need to check that all arrays are of equal length, already done in getListOfValuesPerAsset()
+    ) public view returns (uint256 liquidationValue) {
+        //No need to heck that all arrays are of equal length, already done in getListOfValuesPerAsset()
```
Comment (j-vp) [inline src/AssetRegistry/MainRegistry.sol:489]:
```suggestion
        //No need to check that all arrays are of equal length, already done in getListOfValuesPerAsset()
```

File: src/AssetRegistry/ATokenPricingModule.sol [hunk 5]
```diff
@@ -138,7 +138,7 @@ contract ATokenPricingModule is PricingModule {
         public
         view
         override
-        returns (uint256 valueInUsd, uint256 valueInBaseCurrency, uint256 collFactor, uint256 liqThreshold)
+        returns (uint256 valueInUsd, uint256 valueInBaseCurrency, uint256 collateralFactor, uint256 liquidationFactor)
```
Comment (j-vp) [inline src/AssetRegistry/ATokenPricingModule.sol:141]: add all return variables in documentation

File: src/AssetRegistry/FloorERC1155PricingModule.sol [hunk 6]
```diff
@@ -157,7 +157,7 @@ contract FloorERC1155PricingModule is PricingModule {
         public
         view
         override
-        returns (uint256 valueInUsd, uint256 valueInBaseCurrency, uint256 collFactor, uint256 liqThreshold)
+        returns (uint256 valueInUsd, uint256 valueInBaseCurrency, uint256 collateralFactor, uint256 liquidationFactor)
```
Comment (j-vp) [inline src/AssetRegistry/FloorERC1155PricingModule.sol:160]: add all return variables in documentation

File: src/AssetRegistry/FloorERC721PricingModule.sol [hunk 7]
```diff
@@ -188,12 +188,12 @@ contract FloorERC721PricingModule is PricingModule {
         public
         view
         override
-        returns (uint256 valueInUsd, uint256 valueInBaseCurrency, uint256 collFactor, uint256 liqThreshold)
+        returns (uint256 valueInUsd, uint256 valueInBaseCurrency, uint256 collateralFactor, uint256 liquidationFactor)
```
Comment (j-vp) [inline src/AssetRegistry/FloorERC721PricingModule.sol:191]: add all return variables in documentation

File: src/AssetRegistry/StandardERC20PricingModule.sol [hunk 8]
```diff
@@ -123,7 +123,7 @@ contract StandardERC20PricingModule is PricingModule {
         public
         view
         override
-        returns (uint256 valueInUsd, uint256 valueInBaseCurrency, uint256 collFactor, uint256 liqThreshold)
+        returns (uint256 valueInUsd, uint256 valueInBaseCurrency, uint256 collateralFactor, uint256 liquidationFactor)
```
Comment (j-vp) [inline src/AssetRegistry/StandardERC20PricingModule.sol:126]: add all return variables in documentation

File: src/AssetRegistry/UniswapV2PricingModule.sol [hunk 9]
```diff
@@ -124,7 +124,7 @@ contract UniswapV2PricingModule is PricingModule {
         public
         view
         override
-        returns (uint256 valueInUsd, uint256 valueInBaseCurrency, uint256 collFactor, uint256 liqThreshold)
+        returns (uint256 valueInUsd, uint256 valueInBaseCurrency, uint256 collateralFactor, uint256 liquidationFactor)
```
Comment (j-vp) [inline src/AssetRegistry/UniswapV2PricingModule.sol:127]: add all return variables in documentation

File: src/RiskModule.sol [hunk 10]
```diff
@@ -20,28 +20,23 @@ library RiskModule {
...
         uint256 valueInBaseCurrency;
-        uint256 collFactor;
-        uint256 liqThreshold;
-    }
-
-    struct AssetRisk {
-        address asset;
-        uint16[] assetCollateralFactors;
-        uint16[] assetLiquidationThresholds;
+        uint256 collateralFactor;
```
Comment (j-vp) [inline src/RiskModule.sol:23]: can these be packed?

File: src/Vault.sol [hunk 11]
```diff
@@ -240,69 +234,46 @@ contract Vault {
 
     /**
      * @notice Can be called by authorised applications to increase a margin position.
-     * @param baseCurrency The Base-currency in which the margin position is denominated
+     * @param baseCurrency_ The Base-currency in which the margin position is denominated
      * @param amount The amount the position is increased.
      * @return success Boolean indicating if there is sufficient free margin to increase the margin position
      * @dev The Liquidation Threshold will automatically be updated on every increase of margin,
```
Comment (j-vp) [inline src/Vault.sol:240]: comment can go

File: src/Vault.sol [hunk 12]
```diff
@@ -240,69 +234,46 @@ contract Vault {
...
-    function increaseMarginPosition(address baseCurrency, uint256 amount)
+    function increaseMarginPosition(address baseCurrency_, uint256 amount)
         public
+        view
         onlyAuthorized
         returns (bool success)
     {
-        if (baseCurrency != vault.baseCurrency) {
+        if (baseCurrency_ != baseCurrency) {
             return false;
```
Comment (j-vp) [inline src/Vault.sol:250]: to check if this can be replaced by a revert

File: src/Vault.sol [hunk 13]
```diff
@@ -353,42 +340,21 @@ contract Vault {
      * @dev Requires an unhealthy vault (value / debt < liqThres).
      * Starts the vault auction on the liquidator contract.
      * Increases the life of the vault to indicate a liquidation has happened.
-     * Sets debtInfo todo: needed?
      * Transfers ownership of the proxy vault to the liquidator!
-     * @param liquidationKeeper Addross of the keeper who initiated the liquidation process.
+     * @param liquidationKeeper Address of the keeper who initiated the liquidation process.
      * @return success Boolean returning if the liquidation process is successfully started.
      */
     function liquidateVault(address liquidationKeeper) public onlyFactory returns (bool success, address liquidator_) {
```
Comment (j-vp) [inline src/Vault.sol:347]: todo: rename liquidationkeeper

File: src/Vault.sol (hunk 12 above)
Comment (Thomas-Smets) [reply src/Vault.sol:250]: As just discussed, we'll keep it as it is, since TrustedCreditor might implement custom logic to deal with it.

**Review by j-vp — APPROVED** [review]
lgtm

**Pull request merged**